src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -135,22 +135,16 @@ public Command parseCommand(String userInput) throws ParseException {
         case LinkCommand.COMMAND_WORD:
             return new LinkCommandParser().parse(arguments);
         case AddTaskCommand.COMMAND_WORD:
-            if (!model.getState().equals(SCHEDULE)) {
-                throw new ParseException("Please add tasks in the schedule list.");
-            }
-            return new AddTaskCommandParser().parse(arguments);
-        case "task":
-            if (!model.getState().equals(SCHEDULE)) {
+            if (!model.sameState(SCHEDULE)) {
                 throw new ParseException("Please add tasks in the schedule list.");
             }
             return new AddTaskCommandParser().parse(arguments);
         case DeleteTaskCommand.COMMAND_WORD:
-            if (!model.getState().equals(SCHEDULE)) {
```

Comment:
emm, could you use the refractor tool to globally rename this method to "isSameState"? 
Not very important though

Reply:
okay corrected it